src/Init/Tactics.lean

```diff
@@ -1161,6 +1161,122 @@ syntax (name := pushCast) "push_cast" (config)? (discharger)? (&" only")?
 -/
 syntax (name := normCastAddElim) "norm_cast_add_elim" ident : command
 
+/--
+* `symm` applies to a goal whose target has the form `t ~ u` where `~` is a symmetric relation,
+  that is, a relation which has a symmetry lemma tagged with the attribute [symm].
+  It replaces the target with `u ~ t`.
+* `symm at h` will rewrite a hypothesis `h : t ~ u` to `h : u ~ t`.
+-/
+syntax (name := symm) "symm" (Parser.Tactic.location)? : tactic
+
+section SolveByElim
+
+namespace SolveByElim
+
+/-- Syntax for omitting a local hypothesis in `solve_by_elim`. -/
+syntax erase := "-" term:max
+/-- Syntax for including all local hypotheses in `solve_by_elim`. -/
+syntax star := "*"
+/-- Syntax for adding or removing a term, or `*`, in `solve_by_elim`. -/
+syntax arg := star <|> erase <|> term
+/-- Syntax for adding and removing terms in `solve_by_elim`. -/
+syntax args := " [" SolveByElim.arg,* "]"
+/-- Syntax for using all lemmas labelled with an attribute in `solve_by_elim`. -/
+syntax using_ := " using " ident,*
+
+end SolveByElim
+
+open SolveByElim (args using_)
```

Comment:
This is at the end of the `Tactic` namespace for now, but I think it would be good to add a scoping `section` around this `open`, so that `args` doesn't pollute the namespace when people add more material below this.

Reply:
There is a section for SolveByElim on line 1172.  I'll move it just so the relationship and open is more clear.